### DISCUSSION TITLE
Grouping, grouped placeholder cell custom rendering

### DISCUSSION THREAD

**Discussion by Hideman85** [KevinVandy/material-react-table#367] [Ideas]
I have a case that I do need the place holder cell when grouping to render something.
Take the example of a todo list, you have Status, Task type, Description, Owner. Then the Task type has a custom rendering to be also a link to the object with the task.

| Status | Task type | Desc | Owner |
|---|---|---|---|
| to do | [Review](#) | A task desc | you |
| done | [Review](#) | A completed one | you |

Now grouping by Task type you are having an empty placeholder cell that I would to change the render to have Link.

Today it produce

|Task type|Status|Desc|Owner|
|---|---|---|---|
|Review||||
||to do|A task desc|you|
||done|A completed one|you|

And want

|Task type|Status|Desc|Owner|
|---|---|---|---|
|Review||||
|[Link](#)|to do|A task desc|you|
|[Link](#)|done|A completed one|you|

**Comment by KevinVandy** [comment]
Are you using the `GroupedCell` render override?

**Reply by Hideman85** [reply]
I do, but it does not seems to allow me to render on the children rows 🤔 or I did a mistake somewhere

**Reply by MarkusEllyton** [reply]
I've actually faced a similar issue. If I'm not mistaken, it's pretty hard to get those cells to render with the current implementation. Cells that are in a grouped column, but whose row's columnGroupingId doesn't correspond to the cell's column.id are considered placeholders. These cells end up in this case in `MRT_TableBodyCell.tsx`:

```typescript
{cell.getIsPlaceholder() ? null ...
 ```
Meaning there isn't much to do to make these cells render. 

Also, due to the definition of a placeholder-cell, there can only be one GroupedCell per row/column like shown below. So there currently isn't any way to control the rendering of any cells sharing a column or row with a GroupedCell. 
![placeholder_demo](https://user-images.githubusercontent.com/70698952/218181037-9c9f1098-ce7c-415a-8e74-7646488bd1e9.png)

Does it make sense to optionally replace the null above with a `PlaceholderCell`? Or of course delegate the rendering to `TableBodyCellValue` as is the case for `AggregatedCell` and `GroupedCell`.

**Comment by KevinVandy** [comment]
After seeing a couple of similar conversations this week, do you think if I added a `PlaceholderCell` column option to override the render of these cells, that that would solve your issue?

**Reply by Hideman85** [reply]
Thanks @MarkusEllyton to voice your case with also more details from the sources.

@KevinVandy Yes definitely, having a `PlaceholderCell` at column level would definitely solve the issue 🙏

**Reply by KevinVandy** [reply]
> Thanks @MarkusEllyton to voice your case with also more details from the sources.
> 
> @KevinVandy Yes definitely, having a `PlaceholderCell` at column level would definitely solve the issue 🙏

It was added in yesterday's release

**Reply by Hideman85** [reply]
That is awesome, this works as expect thank you so much 🙂
